src/Controls/src/Core/Border.cs

```diff
@@ -224,5 +263,38 @@ void OnStrokeDashArrayChanged(object? sender, NotifyCollectionChangedEventArgs e
 		{
 			Handler?.UpdateValue(nameof(IBorderStroke.StrokeDashPattern));
 		}
+
+		class WeakStrokeShapeChangedProxy : WeakEventProxy<VisualElement, EventHandler>
```

Comment:
@jonathanpeppers Is this the right type and way to sue? Is there a more magical one that does not need a new type?

If there is no type now, is it possible to make one as I am sure I have seen a bunch of these "simple" proxies.

Reply:
We should add a generic `WeakPropertyChangedProxy` like this one for `INotifyCollectionChanged`:

https://github.com/dotnet/maui/blob/57e186d16df649524d13def2220df62f700fa9b4/src/Controls/src/Core/Internals/WeakEventProxy.cs#L64

I didn't do that in my recent PRs, because they were subscribing to multiple custom events like `InvalidateGradientRequested`, etc.

We should make a generic proxy type for `INotifyPropertyChanged.PropertyChanged` that can be reused everywhere. Also  check if there are places to go ahead and use it.

Reply:
Created a generic `WeakNotifyPropertyChangedProxy`.

Reply:
Also added some unit tests to validate the property changes and dispose the references. The Border stays alive after unsubscribe the proxy. So, need to review it, something else (apart from the brushes) keep the reference.

Reply:
I think the problem might be: https://github.com/dotnet/maui/pull/13793#discussion_r1135598987